sphinxcontrib/needs/templates/permalink.html

```diff
@@ -0,0 +1,68 @@
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <title>sphinx-needs permalink</title>
+
+    <script>
+        function loadJSON(filename, callback) {
+            var xobj = new XMLHttpRequest();
+            xobj.overrideMimeType('application/json');
+            xobj.open('GET', filename, true);
+            xobj.onreadystatechange = function () {
+                if (xobj.readyState == 4 && xobj.status == '200') {
+                    callback(xobj.responseText);
+                }
+            };
+            xobj.send(null);
+        }
+
+        function main() {
+            loadJSON('{{ needs_file }}', function (response) {
+                const needs = JSON.parse(response);
+                const current_version = needs['current_version'];
+                const versions = needs['versions'];
+                const version = versions[current_version];
+                const needs_obj = version['needs'];
+
+                const id = getParameterByName('id');
+                var pathname = new URL(window.location.href).pathname;
+                pathname = pathname.substring(0, pathname.lastIndexOf('{{ permalink_file }}'));
+
+                const keys = Object.keys(needs_obj);
+
+                var docname = 'index';
+
+                keys.forEach((key, index) => {
+                    if (key === id) {
+                        const need = needs_obj[key];
+                        docname = need['docname'];
+                        return;
+                    }
+                });
+
+                window.location.replace(pathname + docname + '.html#' + id);
+            });
+        }
+
+        function getParameterByName(name, url = window.location.href) {
```

Comment:
Just a small hint, by default jQuery is available on Sphinx docs.

So I think some stuff could be solved much easier/shorter with jQuery.
Just for the case, you need to touch some js-parts again.
Otherwise it's fine :)

Reply:
> What do you think, makes this sense?

Totally. With the long break I somehow forgot about this complication ;-)
Thanks for your thorough review 👍 

I changed the computation of the link in `layout.py` to a relative link based on the need docname as you described earlier.
The web browser automatically resolves this to an absolute URL when the web page is rendered.
I tested with a local webserver one time hosting _build/html and another test hosting _build.
It worked.
I also tested opening without web server directly from Windows file explorer. Here the link looked good as well but the permalink page itself does not work because of CORS error when loading needs.json.

So I guess I can get arround more JS "magic" ;-)

Please test for yourself again.

I hope I understood you correctly and it is ok now.
Otherwise just tell me and I hope for your continued patience ;-)